spark/interpreter/src/main/java/org/apache/zeppelin/spark/Utils.java

```diff
@@ -149,7 +149,12 @@ static boolean isSpark2() {
   }
   
   public static String buildJobGroupId(InterpreterContext context) {
-    return "zeppelin-" + context.getNoteId() + "-" + context.getParagraphId();
+    String userIdAndHyphen = "";
```

Comment:
The default value should be `anonymous`.

Reply:
I think that if `context.getAuthenticationInfo()` is null, before rule of jobGroupId is better.

So, I didn't write hyphen in 157 line.
Is it better to add userId always?

Reply:
To me, the user would either be `anonymous` or the real user name. Otherwise there must be some bugs in zeppelin.